test/cpp/end2end/grpclb_end2end_test.cc

```diff
@@ -125,12 +125,20 @@ class CountedService : public ServiceType {
 using BackendService = CountedService<TestServiceImpl>;
 using BalancerService = CountedService<LoadBalancer::Service>;
 
+const char g_kCallCredsMdKey[] = "Balancer should not ...";
+const char g_kCallCredsMdValue[] = "... receive me";
+
 class BackendServiceImpl : public BackendService {
  public:
   BackendServiceImpl() {}
 
   Status Echo(ServerContext* context, const EchoRequest* request,
               EchoResponse* response) override {
+    // Backend should receive the call credentials metadata.
+    auto call_credentials_entry =
+        context->client_metadata().find(g_kCallCredsMdKey);
+    EXPECT_NE(call_credentials_entry, context->client_metadata().end());
```

Comment:
This should probably be `ASSERT_NE()`, because if the check fails, the next line will cause a crash.

If `ASSERT_NE()` won't work in this context, then you can just wrap the next line in a `if (call_credentials_entry != context->client_metadata().end())` conditional.